jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -31,6 +31,7 @@
 import java.lang.annotation.*;
 import java.util.Collection;
 import java.util.HashMap;
+import java.util.HashSet;
```

Comment:
Pls only include this for JAVA_SPEC_VERSION >= 15 to avoid the warning for older versions.

Reply:
Also, these should probably change to >= 16 since getPermittedSubclasses() is introduced in 16.

Reply:
http://cr.openjdk.java.net/~iris/se/16/build/latest/diffsFrom15%2b36/java.base/java/lang/Class.html says it's `since 15`. Should that be retroactive or not?

Reply:
Note jdk15 is no longer supported, and wouldn't build now anyway.

Reply:
Updated the check to version >= 16 then.

Reply:
The javadoc should include `@since 15`, but the code doesn't need to support building for 15. We have a branch for that.

Reply:
I meant, change them all. Where cachedPermittedSubclasses is declared, and around `getPermittedSubclasses()`.
Also pls add the `@since 15` to the javadoc comment.

Reply:
Should the export for `permittedSubclassesImpl` also be moved to jdk 16?

Reply:
I think changing C code is outside the scope of this change since you aren't modifying it already. It's inconsistent but we don't build 15 anyway. Change anything that makes sense in the Java code.
It just seems wrong to include `getPermittedSubclasses()` for 15 when it was introduced in 16.